disk creation error

I have setup terraform 0.11.11 + terraform-provider-proxmox + proxmox-api-go
I can create vm ( qemu )  without disk , with no problem 

When i try to add a disk , terraform aplly  hangs with this error 

resource "proxmox_vm_qemu" "ljrsrv1" {
  name               = "ljrsrv1"
  desc               = "SSO VM"
  target_node        = "${var.vm_node}"
  memory             = 4096
  sockets            = 1
  cores              = 4
  iso = "VMs:iso/ljrsrv1.iso"
  network = {
    id  = 0
    model   = "${var.vm_net_driver}"
    bridge  = "${var.vm_net_bridge}"
  }
  disk = {
    id = 0
    type = "virtio"
    storage = "VMs"
    storage_type = "dir"
    size = "50"
    format = "raw"
  }
}


1 error(s) occurred:
* proxmox_vm_qemu.ljrsrvsso1: 1 error(s) occurred:
* proxmox_vm_qemu.ljrsrvsso1: Cannot create VM disk VMs:101/vm-101-disk-1.raw 

with debug mode : 
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: 2018/12/29 14:32:00 >>>>>>>>>> REQUEST:
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: %!(EXTRA string=POST /api2/json/nodes/lrjhyp1/storage/VMs/content HTTP/1.1
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: Host: 89.31.146.4:8006
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: User-Agent: Go-http-client/1.1
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: Content-Length: 49
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: Accept: application/json
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: Content-Type: application/x-www-form-urlencoded
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: Cookie: PVEAuthCookie=PVE:root@pam:5C278560::YcY/5F982RVfJcrQCOptIQr6terrtsy+ds49L4Lf7eGRd9gjDB1NmmNchEqaln5+vvkt7xfVBfm+ItD1jZCgv9yUmHAulvRgLPw5Zls2USwTJisFrHWRpBTKLZSr6V/4o8v2y9mFySALjCTGc9YbJ8lYpVT5xRZAu67PHCJA4meZbYMNp9jlfBx7AW0hDGZ+czfQIkYqovsvPx4rqA6nZzUCS5JB1ahqPlGeEIirMx6gDWKBog35K3UXKD8UorVO7WdRKQ1CrvA+A89TskN0xuJwKGgD7NMPBQuYalE6W38AZrH7zaIjBtofXWI4e2ukAuAYEIackkgE4Qpnhy9aSQ==
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: Csrfpreventiontoken: 5C278560:5Ms3bpxrWNhthstAxML68Tv+tkg
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: Accept-Encoding: gzip
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: 
2018-12-29T14:32:00.450Z [DEBUG] plugin.terraform-provider-proxmox: filename=101%2Fvm-101-disk-1.raw&size=50&vmid=101)
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: 2018/12/29 14:32:00 <<<<<<<<<< RESULT:
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: %!(EXTRA string=HTTP/1.1 500 unable to parse volume filename '101/vm-101-disk-1.raw'
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: Connection: close
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: Content-Length: 13
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: Cache-Control: max-age=0
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: Content-Type: application/json;charset=UTF-8
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: Date: Sat, 29 Dec 2018 14:32:00 GMT
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: Expires: Sat, 29 Dec 2018 14:32:00 GMT
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: Pragma: no-cache
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: Server: pve-api-daemon/3.0
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: 
2018-12-29T14:32:00.485Z [DEBUG] plugin.terraform-provider-proxmox: {"data":null})
2018/12/29 14:32:00 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

--- a/proxmox/client.go
+++ b/proxmox/client.go
@@ -473,5 +473,13 @@ func getStorageAndVolumeName(
 ) (storageName string, diskName string) {
 	storageAndVolumeName := strings.Split(fullDiskName, separator)
 	storageName, volumeName := storageAndVolumeName[0], storageAndVolumeName[1]
+
+	// when disk type is dir, volumeName is `file=local:100/vm-100-disk-0.raw`
+	re := regexp.MustCompile(`\d+/(?P<filename>\S+.\S+)`)
+	match := re.FindStringSubmatch(volumeName)
+	if len(match) == 2 {
+		volumeName = match[1]
+	}
+
 	return storageName, volumeName
 }

